IBPSA/Media/Examples/SteamDerivativeCheck.mo

```diff
@@ -0,0 +1,71 @@
+within IBPSA.Media.Examples;
+model SteamDerivativeCheck "Model that tests the derivative implementation"
+  extends Modelica.Icons.Example;
+
+   package Medium = IBPSA.Media.Steam "Medim model";
+
+    Modelica.SIunits.Temperature T "Temperature";
+    Modelica.SIunits.SpecificEnthalpy hVapSym "Vapor phase specific enthalpy";
+    Modelica.SIunits.SpecificEnthalpy hVapCod "Vapor phase specific enthalpy";
+    Modelica.SIunits.SpecificHeatCapacity cpSym "Specific heat capacity";
+    Modelica.SIunits.SpecificHeatCapacity cpCod "Specific heat capacity";
+    Modelica.SIunits.SpecificHeatCapacity cvSym "Specific heat capacity";
+    Modelica.SIunits.SpecificHeatCapacity cvCod "Specific heat capacity";
+    constant Real convT(unit="K/s3") = 270
+    "Conversion factor to satisfy unit check";
+initial equation
+     hVapSym = hVapCod;
+     cpSym   = cpCod;
+     cvSym   = cvCod;
+equation
+    T = 273.15+110+convT*time^3;
+    hVapCod=Medium.specificEnthalpy(
+      Medium.setState_pTX(
+         p=1e5,
+         T=T,
+         X=Medium.X_default));
+    assert(abs(hVapCod-hVapSym) < 1E-4 * (1+abs(hVapCod)), "Model has an error");
+    der(hVapCod)=der(hVapSym);
+
+    cpCod=Medium.specificHeatCapacityCp(
+      Medium.setState_pTX(
+         p=1e5,
+         T=T,
+         X=Medium.X_default));
+    der(cpCod)=der(cpSym);
+    assert(abs(cpCod-cpSym) < 1E-4 * (1+abs(cvCod)), "Model has an error");
+
+     cvCod=Medium.specificHeatCapacityCv(
+      Medium.setState_pTX(
+         p=1e5,
+         T=T,
+         X=Medium.X_default));
+    der(cvCod)=der(cvSym);
+    assert(abs(cvCod-cvSym) < 1E-4 * (1+abs(cvCod)), "Model has an error");
```

Comment:
> Model has an error

is not a really good error message. Adding a hint which variable is affected, would be helfpful.